### PR TITLE
=docs #17600 fix incorrect http client example

### LINKED ISSUE #17600: Akka HTTP Client Connection-level (1.0-RC3) fails the documentation code example
The code given in the documentation [here](http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0-RC3/scala/http/client-side/connection-level.html) throws an exception:

```scala
import scala.concurrent.Future
import akka.stream.ActorFlowMaterializer
import akka.stream.scaladsl._
import akka.http.scaladsl.model._
import akka.http.scaladsl.Http
 
implicit val system = ActorSystem()
implicit val materializer = ActorFlowMaterializer()
 
val connectionFlow: Flow[HttpRequest, HttpResponse, Future[Http.OutgoingConnection]] =
  Http().outgoingConnection("http://akka.io")
val responseFuture: Future[HttpResponse] =
  Source.single(HttpRequest(uri = "/"))
    .via(connectionFlow)
    .runWith(Sink.head)
```

The exception details and trace are as follows:

```
[ERROR] [05/28/2015 23:34:30.669] [default-akka.actor.default-dispatcher-3] [ActorSystem(default)] Outgoing request stream error
akka.http.scaladsl.model.IllegalUriException: Illegal URI host: Invalid input ':', expected lower-reg-name-char, UPPER_ALPHA, '%' or 'EOI' (line 1, column 5): http://akka.io
    ^
	at akka.http.scaladsl.model.IllegalUriException$.apply(ErrorInfo.scala:40)
	at akka.http.scaladsl.model.Uri$.fail(Uri.scala:748)
	at akka.http.impl.model.parser.UriParser.fail(UriParser.scala:61)
	at akka.http.impl.model.parser.UriParser.parseHost(UriParser.scala:51)
	at akka.http.scaladsl.model.Uri$Host$.apply(Uri.scala:350)
	at akka.http.scaladsl.model.headers.Host$.apply(headers.scala:95)
	at akka.http.impl.engine.rendering.HttpRequestRendererFactory$HttpRequestRenderer.renderHeaders$1(HttpRequestRendererFactory.scala:110)
	at akka.http.impl.engine.rendering.HttpRequestRendererFactory$HttpRequestRenderer.onPush(HttpRequestRendererFactory.scala:139)
	at akka.http.impl.engine.rendering.HttpRequestRendererFactory$HttpRequestRenderer.onPush(HttpRequestRendererFactory.scala:42)
	at akka.stream.impl.fusing.OneBoundedInterpreter$$anon$1.run(Interpreter.scala:436)
	at akka.stream.impl.fusing.OneBoundedInterpreter$State$class.progress(Interpreter.scala:245)
	at akka.stream.impl.fusing.OneBoundedInterpreter$$anon$1.progress(Interpreter.scala:434)
	at akka.stream.impl.fusing.OneBoundedInterpreter.akka$stream$impl$fusing$OneBoundedInterpreter$$execute(Interpreter.scala:580)
	at akka.stream.impl.fusing.OneBoundedInterpreter$State$class.execute(Interpreter.scala:241)
	at akka.stream.impl.fusing.OneBoundedInterpreter$EntryState.execute(Interpreter.scala:666)
	at akka.stream.stage.AbstractStage.enterAndPush(Stage.scala:64)
	at akka.stream.impl.fusing.BatchingActorInputBoundary$$anonfun$upstreamRunning$1.applyOrElse(ActorInterpreter.scala:143)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at akka.stream.impl.SubReceive.apply(Transfer.scala:16)
	at akka.stream.impl.SubReceive.apply(Transfer.scala:12)
	at scala.PartialFunction$class.applyOrElse(PartialFunction.scala:123)
	at akka.stream.impl.SubReceive.applyOrElse(Transfer.scala:12)
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:170)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:467)
	at akka.stream.impl.fusing.ActorInterpreter.aroundReceive(ActorInterpreter.scala:352)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
	at akka.actor.ActorCell.invoke(ActorCell.scala:487)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:238)
	at akka.dispatch.Mailbox.run(Mailbox.scala:220)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

### PR DESCRIPTION
**Pull request by sirthias**
Fixes #17600.

/cc @jrudolph

### CODE REVIEW

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpClientExampleSpec.scala [hunk 1]
```diff
@@ -64,7 +64,7 @@ class HttpClientExampleSpec extends WordSpec with Matchers {
     implicit val materializer = ActorFlowMaterializer()
 
     val responseFuture: Future[HttpResponse] =
-      Http().singleRequest(HttpRequest(uri = "http://akka.io"))
+      Http().singleRequest(HttpRequest(uri = "akka.io"))
```
Comment (jrudolph) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpClientExampleSpec.scala]: I think something broke here...

Comment (drewhk) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpClientExampleSpec.scala]: wdym?

Comment (jrudolph) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpClientExampleSpec.scala]: This change breaks the test as "akka.io" is then interpreted as a relative URI.

Comment (sirthias) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpClientExampleSpec.scala]: Indeed. Too-Aggressive-S&R-Error. Will fix.

**Comment by sirthias** [comment]
Ok, fixed. (And added commit containing a review improvement rightfully suggested by @jrudolph.)

**Comment by sirthias** [comment]
TlsSpec.

**Comment by drewhk** [comment]
LGTM

**Pull request merged**